scripts/upgrade-via-npm.sh

```
@@ -1,652 +1,845 @@
+    if [ -n "$_PREV_RELOAD_MODE" ]; then
+        openclaw config set gateway.reload.mode "$_PREV_RELOAD_MODE" 2>/dev/null || true
+    else
+        openclaw config unset gateway.reload.mode 2>/dev/null || true
+    fi
+    _PREV_RELOAD_MODE=""   # 防止重复执行
+}
+
+rollback_plugin_dir() {
+    local reason="${1:-未知原因}"
+    if [ -n "$BACKUP_DIR" ] && [ -d "$BACKUP_DIR/$PLUGIN_ID" ]; then
+        rm -rf "$EXTENSIONS_DIR/$PLUGIN_ID" 2>/dev/null || true
```

Shellcheck (0.11.0)

[warning] Use "${var:?}" to ensure this never expands to / .

(SC2115)